src/compiler/parser.ts

```diff
@@ -2431,6 +2431,18 @@ namespace ts {
 
         function parseJSDocType(): TypeNode {
             scanner.setInJSDocType(true);
+            const moduleSpecifier = parseOptionalToken(SyntaxKind.ModuleKeyword);
+            if (moduleSpecifier) {
+                const moduleTag = createNode(SyntaxKind.JSDocNamepathType, moduleSpecifier.pos) as JSDocNamepathType;
+                const terminators = [SyntaxKind.CloseBraceToken, SyntaxKind.EndOfFileToken, SyntaxKind.CommaToken, SyntaxKind.CloseParenToken, SyntaxKind.WhitespaceTrivia];
+                while (terminators.indexOf(token()) < 0) {

```

Comment:
```suggestion
                terminate: while (true) {
                    switch(token()) {
                        case SyntaxKind.CloseBraceToken: // etc
                            break terminate;
                    }
```

Reply:
I love it, this is the only project I've ever had PR suggestions where people recommend `goto`.

Reply:
Gotta go fast!